alfresco-transform-tika/alfresco-transform-tika/src/main/java/org/alfresco/transformer/tika/parsers/ExifToolParser.java

```diff
@@ -0,0 +1,260 @@
+                    extractMetadata(out, metadata);
+                }
+            } else {
+                ignoreStream(err);
+
+                if (outputFromStdOut) {
+                    extractOutput(out, xhtml);
+                } else {
+                    ignoreStream(out);
+                }
+            }
+        } finally {
+            try {
+                process.waitFor();
+            } catch (InterruptedException ignore) {
+            }
+        }
+
+        // Grab the output if we haven't already
+        if (!outputFromStdOut) {
+            extractOutput(new FileInputStream(output), xhtml);
+        }
+    }
+
+    /**
+     * Adapted from {@link org.apache.tika.parser.external.ExternalParser}<p>
+     * Starts a thread that extracts the contents of the standard output
+     * stream of the given process to the given XHTML content handler.
+     * The standard output stream is closed once fully processed.
+     *
+     * @param process process
+     * @param xhtml XHTML content handler
+     * @throws SAXException if the XHTML SAX events could not be handled
+     * @throws IOException if an input error occurred
+     */
+    private void extractOutput(InputStream stream, XHTMLContentHandler xhtml) throws SAXException, IOException {
+        try (Reader reader = new InputStreamReader(stream, UTF_8)) {
+            xhtml.startDocument();
+            xhtml.startElement("p");
+            char[] buffer = new char[1024];
+            for (int n = reader.read(buffer); n != -1; n = reader.read(buffer)) {
+                xhtml.characters(buffer, 0, n);
+            }
+            xhtml.endElement("p");
+            xhtml.endDocument();
+        }
+    }
+
+    /**
+     * Adapted from {@link org.apache.tika.parser.external.ExternalParser}<p>
+     * Starts a thread that sends the contents of the given input stream
+     * to the standard input stream of the given process. Potential
+     * exceptions are ignored, and the standard input stream is closed
+     * once fully processed. Note that the given input stream is <em>not</em>
+     * closed by this method.
+     *
+     * @param process process
+     * @param stream input stream
+     */
+    private void sendInput(final Process process, final InputStream stream) {
+        Thread t = new Thread() {
+            public void run() {
+                OutputStream stdin = process.getOutputStream();
+                try {
+                    IOUtils.copy(stream, stdin);
+                } catch (IOException e) {
+                }
+            }
+        };
+        t.start();
+        try {
+            t.join();
+        } catch (InterruptedException ignore) {
+        }
+    }
+
+    /**
+     * Adapted from {@link org.apache.tika.parser.external.ExternalParser}<p>
+     * Starts a thread that reads and discards the contents of the
+     * standard stream of the given process. Potential exceptions
+     * are ignored, and the stream is closed once fully processed.

```

Comment:
Should we log "potential exceptions" (eg. as DEBUG or TRACE) ?

Reply:
If the issue we currently have is fixed by Tika, we should remove this whole class and just use the standard functionality of ExternalParser. This only modification I have made to these methods is to remove a line of code that was producing the bug (which was at approx L146: extractMetadata(err, metadata);
I don't think we should be adding additional logging now, with expectation of removing this class in the future.